src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -194,9 +263,35 @@ private void WriteRawValueCore(ReadOnlySpan<byte> utf8Json, bool skipInputValida
             }
 
             utf8Json.CopyTo(output.Slice(BytesPending));
-            BytesPending += len;
+            BytesPending += (int)len;
```

Comment:
Arguably, this should be a `checked` cast and should have happened much earlier in the validation stage.

Reply:
Added check in earlier validation stage.